examples/applications/gemcutter/models/user.rb

```diff
@@ -1,10 +1,20 @@
 class User < ActiveRecord::Base
-  has_many :rubygems, :through    => :ownerships,
-                      :order      => "name ASC",
-                      :conditions => { 'ownerships.approved' => true }
-  has_many :subscribed_gems, :through => :subscriptions,
-                             :source  => :rubygem,
-                             :order   => "name ASC"
+  if ActiveRecord::VERSION::MAJOR >= 4
+    has_many :rubygems,
+      lambda { where(:ownerships => { :approved => true }).order(:name => :asc) },
+      :through => :ownerships
+    has_many :subscribed_gems,
+      lambda { order(:name => :asc) },
+      :through => :subscriptions,
+      :source  => :rubygem
+  else
+    has_many :rubygems, :through    => :ownerships,
+                        :order      => "name ASC",
+                        :conditions => { 'ownerships.approved' => true }
+    has_many :subscribed_gems, :through => :subscriptions,
+                               :source  => :rubygem,
+                               :order   => "name ASC"
+  end
```

Comment:
This is the sort of version switching I was dreading having to do... well done, thank you :)